### PR TITLE
[CS-59][GREEN-13] Connect Killbill to subscription service

### PR DESCRIPTION
**Pull request by Tbilgere**
-Added ClientBase with configs
-Added Implentation of ClientBase for subscription service
-Updated EmailNotifactionListener to call to Subscription service client instead of sending email
-Added tests for subscription service implentation
-Added mock account and invoice classes for use in testing
-Updated Pom as needed

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -126,6 +126,25 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.w3c</groupId>
+            <artifactId>dom</artifactId>
+            <version>2.3.0-jaxb-1.0.6</version>
+        </dependency>
+        <dependency>
+            <groupId>com.github.tomakehurst</groupId>
+            <artifactId>wiremock</artifactId>
+            <version>1.57</version>
```
Comment (josephlbarnett) [inline pom.xml]: test scope this?

File: src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java [hunk 2]
```diff
@@ -102,6 +107,11 @@ public EmailNotificationListener(final OSGIKillbillClock clock, final OSGIKillbi
         this.clock = clock;
         this.emailSender = new EmailSender(configProperties, logService);
         this.templateRenderer = new TemplateRenderer(new MustacheTemplateEngine(), new ResourceBundleFactory(killbillAPI.getTenantUserApi(), logService), killbillAPI.getTenantUserApi(), logService);
+
+        String subscriptionAddress = configProperties.getString(SUBSCRIPTION_SERVICE_ADDRESS);
+        Preconditions.checkArgument(subscriptionAddress != null, String.format("Cannot find propert %s", SUBSCRIPTION_SERVICE_ADDRESS));
```
Comment (josephlbarnett) [inline src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java]: `property`

File: src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java [hunk 3]
```diff
@@ -193,11 +204,12 @@ private void sendEmailForCancelledSubscription(final Account account, final ExtB
...
-            final EmailContent emailContent = subscription.getState() == Entitlement.EntitlementState.CANCELLED ?
-                    templateRenderer.generateEmailForSubscriptionCancellationEffective(account, subscription, context) :
-                    templateRenderer.generateEmailForSubscriptionCancellationRequested(account, subscription, context);
-            sendEmail(account, emailContent, context);
+        if (subscription != null ) { //&& !muteEmailForObject(subscription.getBundleId(), ObjectType.BUNDLE, context)) {
+            //final EmailContent emailContent = subscription.getState() == Entitlement.EntitlementState.CANCELLED ?
+              //      templateRenderer.generateEmailForSubscriptionCancellationEffective(account, subscription, context) :
+               //     templateRenderer.generateEmailForSubscriptionCancellationRequested(account, subscription, context);
+            //sendEmail(account, emailContent, context);
+            subscriptionClient.sendEmailRequest("Subscription_Canceled", account, subscriptionId.toString());
```
Comment (josephlbarnett) [inline src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java]: do we need to differentiate between Effective and Requested ?

**Review by josephlbarnett — APPROVED** [review]

File: src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java (hunk 3 above)
Comment (Tbilgere) [reply src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java]: That is a good question I'm not sure what the difference is.

File: pom.xml (hunk 1 above)
Comment (pklock) [reply pom.xml]: we already have this in corporate pom as test scope, shouldn't need to specify version here.

**Review by pklock — APPROVED** [review]

**Pull request merged**